src/libNode/MicroBlockPreProcessing.cpp

```diff
@@ -186,24 +266,44 @@ bool Node::OnNodeMissingTxns(const bytes& errorMsg, const unsigned int offset,
 
   std::vector<Transaction> txns;
 
-  const std::unordered_map<TxnHash, TransactionWithReceipt>&
-      processedTransactions = (epochNum == m_mediator.m_currentEpochNum)
-                                  ? t_processedTransactions
-                                  : m_processedTransactions[epochNum];
+  lock_guard<mutex> g(m_mutexCreatedTransactions);
```

Comment:
Maybe can scope this mutex to exclude `SendMessage` below, in case multiple backups are fetching the missing txns